check/requiretls/requiretls.go

```diff
@@ -0,0 +1,26 @@
+package requiretls
+
+import (
+	"github.com/foxcpp/maddy/check"
+	"github.com/foxcpp/maddy/exterrors"
+	"github.com/foxcpp/maddy/module"
+)
+
+func requireTLS(ctx check.StatelessCheckContext) module.CheckResult {
+	if ctx.MsgMeta.SrcTLSState.HandshakeComplete {
```

Comment:
I expected `SrcTLSState` to be a pointer, so it could have a `nil` value. Checking the `HandshakeComplete` boolean works too, but I wonder if it might be better to refactor so it can be a pointer instead. WDYT?

Reply:
No objection. Though, you might as well refactor it in the go-smtp, since maddy simply copies the field from ConnectionState struct passed to it.